### PR TITLE
DIAC-419 fixing aip pay later bug

### PR DESCRIPTION
**Pull request by nilay913**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DIAC-419

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/PaymentStateHandler.java [hunk 1]
```diff
@@ -83,28 +85,40 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
...
                 Optional<HelpWithFeesOption> helpWithFeesOption = asylumCase.read(HELP_WITH_FEES_OPTION, HelpWithFeesOption.class);
                 String paAppealTypePaymentOption = asylumCase.read(PA_APPEAL_TYPE_AIP_PAYMENT_OPTION, String.class).orElse("");
-
-                if (paAppealTypePaymentOption.equals(PAY_LATER)) {
-                    return new PreSubmitCallbackResponse<>(asylumCase, APPEAL_SUBMITTED);
-                } else if (isRemissionExistsAip(remissionOption, helpWithFeesOption)) {
-                    return new PreSubmitCallbackResponse<>(asylumCase, PENDING_PAYMENT);
-                } else {
-                    return new PreSubmitCallbackResponse<>(asylumCase, APPEAL_SUBMITTED);
+                State currentState = callback.getCaseDetails().getState();
```
Comment (ikirsanov) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/PaymentStateHandler.java:97]: Maybe we can move this line before if (isDlrmFeeRemission)

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/PaymentStateHandler.java [hunk 2]
```diff
@@ -83,28 +85,40 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
...
+                    state = currentState;
                 }
+                return new PreSubmitCallbackResponse<>(asylumCase, state);
             } else {
                 return decideAppealState(appealType, isPaymentStatusPendingOrFailed, asylumCase);
             }
-
         } else {
+            if (isAipJourney) {
+                String paAppealTypePaymentOption = asylumCase.read(PA_APPEAL_TYPE_AIP_PAYMENT_OPTION, String.class).orElse("");
```
Comment (ikirsanov) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/PaymentStateHandler.java:112]: this one too

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/PaymentStateHandler.java [hunk 3]
```diff
@@ -83,28 +85,40 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
...
+                State currentState = callback.getCaseDetails().getState();
```
Comment (ikirsanov) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/PaymentStateHandler.java:113]: this one as well

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/PaymentStateHandler.java [hunk 4]
```diff
@@ -83,28 +85,40 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
...
+                List<State> invalidStates = List.of(APPEAL_STARTED, APPEAL_STARTED_BY_ADMIN, PENDING_PAYMENT, ADJOURNED);
```
Comment (ikirsanov) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/PaymentStateHandler.java:114]: this ones too

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Review by jamiephillipshmcts — APPROVED** [review]

**Review by JakeCohenSol — APPROVED** [review]

**Pull request merged**